### ISSUE TITLE
Mac OSX Catalina

### ISSUE THREAD

**Issue by Zman2342** [quodlibet/quodlibet#3267]
Hi,
 Will There be a version of this that works with the new OS?

lzinn@nyc.rr.com

**Comment by lazka** [comment]
What doesn't work with catalina? I'm still on 10.14 on my test machine.

**Comment by Wyzard256** [comment]
It seems fine on Catalina to me, using version 4.2.1 installed via Homebrew ("brew cask install quodlibet").

**Comment by vinu91** [comment]
After upgrading to Catalina, QuodLibet was throwing 'Operation Not Permitted' error when playing songs from the library. It went away after I rebuilt the library (Preferences > Rebuild Library).

Not sure if the original poster was also facing the same issue.

**Comment by tmikaeld** [comment]
Same for me, not sure why this happens but it's incredibly annoying.

I think it's related to the new Catalina file-access-permissions API, where we get a popup asking for permissions to access the disk or network share.

**Comment by lazka** [comment]
Interesting, thanks. Maybe because gstreamer tries to access the file in its own thread where macOS can't create a popup? Maybe we should just try to access the file before passing it to gstreamer.

**Comment by elelay** [comment]
Hi, similar problem with [gPodder](https://github.com/gpodder/gpodder/issues/698) where we can't access the podcast database when it's not in the default location.

OP provides a link to a duplicati issue where a workaround of giving /bin/bash full disk access worked. The safer fix they chose was to rewrite the launcher from a bash script to a native program.

**Comment by lazka** [comment]
Thanks!

**Comment by lazka** [comment]
A bit more info from jralls: https://gitlab.gnome.org/GNOME/gtk-mac-bundler/issues/1#note_638442

**Comment by elelay** [comment]
Thanks, I'll try the https://gitlab.gnome.org/GNOME/gtk-mac-bundler/blob/master/examples/python-launcher.c

**Comment by tmikaeld** [comment]
I'm unable to use the VLC trick now, anyone have a workaround to get QuodLibet working?

**Comment by lazka** [comment]
This should work: https://gitlab.gnome.org/GNOME/gimp/issues/3710#note_630890

**Comment by lazka** [comment]
Fair enough, but that's all I have :)

**Comment by elelay** [comment]
@lazka I've a working native launcher in https://github.com/gpodder/gpodder-osx-bundle/tree/native-launcher.
 1. the native launcher runs a first python script (launcher.py) to set environment variables and run openssl commands from a first python script (easier than in C)
 2. the python script exec's the original main python script (`Contents/Resources/bin/gpodder` in my case); it has to be exec'ed and not run in-process using `runpy` for the environment variables to be picked by all native libraries and dyld.

**Comment by lazka** [comment]
Nice!

**Comment by beepboop1234** [comment]
Any update on this? Same error. The link previously mentioned by elelay gives a 404 and currently QL can't find or play any music in my library.

**Comment by elcste** [comment]
Another point of data: it's working fine for me on 10.15.3 (I haven't updated to 10.15.4 yet). I've used it a system I upgraded from 10.14 as well as a clean install. There are a few new bugs (<https://github.com/quodlibet/quodlibet/issues/3316>, <https://github.com/quodlibet/quodlibet/issues/3322>), but presumably it's launching for the people who reported those bugs too.

**Comment by mundschenk-at** [comment]
I can't access any network volumes on 10.15.4 using brew-installed Ex Falso 4.3.0 - they simply don't show up in the Volumes folder.

**Comment by mundschenk-at** [comment]
While it does not look like anyone much cares about macOS here anymore (currently, there are not even builds for 4.4.0, contrary to other platforms, it looks like the app needs to include some configuration keys in its plist: https://developer.apple.com/documentation/bundleresources/information_property_list/protected_resources

**Comment by elelay** [comment]
@mundschenk-at can you install https://github.com/gpodder/gpodder/releases/download/3.10.17/macOS-gPodder-3.10.17.zip and confirm that you can access a network volume? To do that, subscribe to https://privatecitizen.press/index.xml download the latest episode and right-click/Send To/Local Folder and select a folder on your network drives.

This way we know that the binary launcher works and could be adapted for quodlibet.

**Comment by mundschenk-at** [comment]
> @mundschenk-at can you install https://github.com/gpodder/gpodder/releases/download/3.10.17/macOS-gPodder-3.10.17.zip and confirm that you can access a network volume? To do that, subscribe to https://privatecitizen.press/index.xml download the latest episode and right-click/Send To/Local Folder and select a folder on your network drives.

Yes, that does work. Both network volumes and `Downloads`, `Desktop` et al. are visible and the app requests permission when I click on them.

**Comment by elelay** [comment]
ok, thanks for the confirmation.

**Comment by MikeiLL** [comment]
Came across this issue on `v4.4.0` on osX `v12.3.1` Monterey and was able to solve by:

  1. In Quodlibet > Preferences > Library: Remove external drive directory
  2. Rescan the library
  3. Add the external directory back
  4. Rescan library again
 

Glad to have discovered [gPodder](https://github.com/gpodder/gpodder-osx-bundle)
